### ISSUE TITLE
Molerat is confused by transparency in nested elements

### ISSUE THREAD

**Issue by colons** [data61/wcag-zoo#16]
Molerat will, with an entirely default configuration, consider the following html file to be 'ok':

```html
<!doctype html>
<html>
  <head>
    <title>wcag zoo contrast testing</title>
    <style>
      div {
        background-color: rgb(34, 34, 34);
        color: rgba(255, 255, 255, 0.8);
      }
    </style>
  </head>
  <body>
    <div>woof</div>
  </body>
</html>
```

This, on the other hand, is failed due to inadequate contrast ratios:

```html
<!doctype html>
<html>
  <head>
    <title>wcag zoo contrast testing</title>
    <style>
      div {
        background-color: rgb(34, 34, 34);
        color: rgba(255, 255, 255, 0.8);
      }
    </style>
  </head>
  <body>
    <div>
      <p>woof</p>
    </div>
  </body>
</html>
```

We are told:

```
ERROR - Insufficient contrast (1.31) for text at element - /html/body/p
    Computed rgb values are == Foreground [1, 2, 3] / Background [34, 34, 34]
    Text was:         woof
    Colored text was: woof
    Computed font-size was: 10 normal (normal)
   Hint: Increase the contrast of this text to fix this error
```
